lib/python/rose/config.py

```diff
@@ -1307,6 +1308,36 @@ def load_with_opts(self, source, node=None, more_keys=None,
             return node, config_map
         return node
 
+    def load_defines(self, defines, node=None):
+        """Read configuration from a list of command line defines.
+
+        Arguments:
+            defines (list): A list of [SECTION]KEY=VALUE items.
+            node (ConfigNode - optional): A ConfigNode object if specified,
+                otherwise one is created.
+
+        Returns:
+            ConfigNode: A new ConfigNode object.
+        """
+        # N.B. In theory, we should write the values in "defines" to
+        # "node" directly. However, the values in "defines" may contain
```

Comment:
Perhaps extend the `load` method to also accept a list of defines as a source to load? Then, the logic from `load_defines` could be moved inside the `_get_file_and_name` private method (could also use StringIO instead of TemporaryFile there).